### PR TITLE
Fix bug in ChunkedSliceOutput.getRetainedBytes

### PR DESCRIPTION
**Pull request by shenh062326**
In our production environment, worker OOM occasionally occurs, but the reserve memory is relatively small. After analysis, one of the reasons is that the bufferPool in ChunkedSliceOutput.chunkSupplier are not counted.

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java [hunk 1]
```diff
@@ -365,6 +367,9 @@ public ChunkSupplier(int minChunkSize, int maxChunkSize)
         public void reset()
         {
             bufferPool.addAll(0, usedBuffers);
+            for (byte[] b : bufferPool) {
+                bufferPollSize += SizeOf.sizeOf(b);
```
Comment (v-jizhang) [inline presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java:371]: Can b.length be used here? b is not null.

File: presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java [hunk 2]
```diff
@@ -377,10 +382,16 @@ public void reset()
             }
             else {
                 buffer = bufferPool.remove(0);
+                bufferPollSize -= SizeOf.sizeOf(buffer);
```
Comment (v-jizhang) [inline presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java:385]: Same here buffer.length.

File: presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java (hunk 1 above)
Comment (shenh062326) [reply presto-orc/src/main/java/com/facebook/presto/orc/ChunkedSliceOutput.java:371]: SizeOf.sizeOf(b)=(array == null)? 0: sizeOfByteArray(array.length)
sizeOfByteArray(array.length)=ARRAY_BYTE_BASE_OFFSET + (((long) ARRAY_BYTE_INDEX_SCALE) * length)

It seems that SizeOf.sizeOf(b) is more accurate than b.length.

**Review by v-jizhang — APPROVED** [review]
LGTM

**Comment by rohanpednekar** [comment]
Hi @shenh062326, Do you think you can rebase this PR? We would like to get it reviewed and merged. Please let me know if you need any additional help. Thanks!

**Pull request closed without merging**